homeassistant/components/monzo/config_flow.py

```diff
@@ -33,7 +35,13 @@ async def async_step_await_approval_confirmation(
     ) -> ConfigFlowResult:
         """Wait for the user to confirm in-app approval."""
         if user_input is not None:
-            return self.async_create_entry(title=DOMAIN, data={**self.oauth_data})
+            if not self.reauth_entry:
+                return self.async_create_entry(title=DOMAIN, data={**self.oauth_data})
+            self.hass.config_entries.async_update_entry(
+                self.reauth_entry, data={**self.reauth_entry.data, **self.oauth_data}
+            )
+            await self.hass.config_entries.async_reload(self.reauth_entry.entry_id)
+            return self.async_abort(reason="reauth_successful")
```

Comment:
There's a helper for this, can't remember the name, but I think someone already applied it at Withings

Reply:
Funnily enough, I copied that code from [Withings](https://github.com/home-assistant/core/blob/b8c5dcaeef2a668fb976d2efe8f0f857a02e9d0e/homeassistant/components/withings/config_flow.py#L71-L75) 😆

Reply:
LOL